x-pack/plugins/apm/server/lib/helpers/create_es_client/call_client_with_debug.ts

```diff
@@ -27,12 +28,12 @@ export async function callClientWithDebug({
   operationName: string;
   params: Record<string, any>;
   debug: boolean;
-  request: KibanaRequest;
+  request: KibanaRequest & { _debugQueries?: Array<Record<string, any>> };
```

Comment:
Rather than adding a `_debugQueries` property (which are awkward/impossible with array/string/boolean etc response types), what do you think about wrapping the response if `_debugQueries` set? Something like `{ queries: [], response: ... }`. We can then opaquely unwrap in `useFetcher()`.

Reply:
Hmm.. not following

Reply:
Okay, I get it now - but still not sure where you want to store `_debugQueries` to go from an es request is made until the response happens.